src/main/perl/Fetch.pm

```diff
@@ -183,6 +183,7 @@ sub fetchProfile
     %current = $self->current($profile, %previous);
     if ($self->process_profile("$profile", %current) == $ERROR) {
         $self->error("Failed to process profile for $self->{PROFILE_URL}");
+        $current{cid}->cancel();
```

Comment:
the `local $SIG{__DIE__}` code a few lines back also cancels the other 2 filewriter instances. should this be done here too? (it's mainly a precaution i guess, the previous is only overwritten a few lines below, so that one should be untouched).

anyway, nice catch!